identity-iota/src/credential/validation_options.rs

```diff
@@ -0,0 +1,130 @@
+    self.latest_issuance_date = Some(timestamp);
+    self
+  }
+
+  /// Set options which affect the verification of the signature on the credential.
+  pub fn verifier_options(mut self, options: VerifierOptions) -> Self {
+    self.verifier_options = options;
+    self
+  }
+}
+
+/// Declares how credential subjects must relate to the presentation holder during validation.
+/// See [`PresentationValidationOptions::subject_holder_relationship()`].
+///
+/// See also the [Subject-Holder Relationship](https://www.w3.org/TR/vc-data-model/#subject-holder-relationships) section of the specification.
+// Need to use serde_repr to make this work with duck typed interfaces in the Wasm bindings.
+#[derive(Debug, Clone, Copy, serde_repr::Serialize_repr, serde_repr::Deserialize_repr)]
+#[repr(u8)]
+pub enum SubjectHolderRelationship {
+  /// The holder must always match the subject on all credentials, regardless of their [`nonTransferable`](https://www.w3.org/TR/vc-data-model/#nontransferable-property) property.
+  /// This is the variant returned by [Self::default](Self::default()) and the default used in
+  /// [`PresentationValidationOptions`].
+  AlwaysSubject = 0,
+  /// The holder must match the subject only for credentials where the [`nonTransferable`](https://www.w3.org/TR/vc-data-model/#nontransferable-property) property is `true`.
+  SubjectOnNonTransferable = 1,
+  /// Declares that the subject is not required to have any kind of relationship to the holder.  
+  Any = 2,
+}
+
+impl Default for SubjectHolderRelationship {
+  fn default() -> Self {
+    Self::AlwaysSubject
+  }
+}
+
+/// Declares when validation should return if an error occurs.
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+pub enum FailFast {
+  /// Return all errors that occur during validation.
+  AllErrors,
+  /// Return after the first error occurs.
+  FirstError,
+}
+
+/// Options to declare validation criteria for validation methods such as
+/// [`PresentationValidator::full_validation`](super::PresentationValidator::full_validation()) and
+/// [`Resolver::verify_presentation`](crate::tangle::Resolver::verify_presentation()).
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct PresentationValidationOptions {
+  /// Options which affect the validation of *all* credentials in the presentation.
+  #[serde(default)]
+  pub shared_validation_options: CredentialValidationOptions,
+  /// Options which affect the verification of the signature on the presentation.
+  #[serde(default)]
+  pub presentation_verifier_options: VerifierOptions,
+  /// Declares how the presentation's credential subjects must relate to the holder.
+  #[serde(default)]
+  pub subject_holder_relationship: SubjectHolderRelationship,
+}
+
+impl PresentationValidationOptions {
+  /// Constructor that sets all options to their defaults.
+  pub fn new() -> Self {
+    Self::default()
+  }
```

Comment:
The `Default` implementation should call `new()`, not the other way around.

Reply:
Even when `Default` is derived?

Reply:
Depends on the struct I guess? In this case there is a specific default value that is not derived though.

Reply:
The default is derived [See this line](https://github.com/iotaledger/identity.rs/pull/599/files/6b8e924c6ea6b7aebaa92ca22b73b21481b6b1e7#diff-96a4fb255e2e3ec6dce7c70842f69fbd70a6875cdf223b353e4e8dc231d0e482R93)

Reply:
Ah, my bad. I was confused by the `Default` implementation for `SubjectHolderRelationship` or something.